src/store/actions/payment.js

```diff
@@ -516,24 +517,33 @@ export const putBalanceProof = (message, payment) => async (
   }
 };
 
-export const putNonClosingBalanceProof = (message, payment) => async (
-  dispatch,
-  getState,
-  lh
-) => {
+export const putNonClosingBalanceProof = (
+  message,
+  payment,
+  ltToPack = null
+) => async (dispatch, getState, lh) => {
   const { getAddress } = ethers.utils;
-  const dataToSign = getDataToSignForNonClosingBalanceProof(message);
+  // LT may only be present when we sign after a Reveal Secret 7
+  // In other cases, we only sign a BP
+  // The data is abstracted to this const to avoid repeated code
+  const dataForPack = ltToPack || message;
+  const dataToSign = ltToPack
+    ? getDataToSignForLockedTransfer(
+        dataForPack,
+        MessageTypeID.UPDATE_BALANCE_PROOF
+      )
+    : getDataToSignForNonClosingBalanceProof(dataForPack);
```

Comment:
Looking this a bit closer, we could inject the packed data instead of two mutually exclusive params 'message' and 'ltToPack'

Reply:
I'm not understanding your idea, could you explain it again?

Reply:
I don't like to have two params that are someway the same. but I don-t think it-s too relevant right now